lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -33,23 +33,40 @@
 #include "EnumToString.h"
 #include "SessionStateEnforcer.h"
 
+std::shared_ptr<magma::SessionStateEnforcer> conv_session_enforcer;
 namespace magma {
-// temp routine
-void call_back_void_upf(grpc::Status, magma::UpfRes response) {
-  // do nothinf but to only passing call back
+void call_back_upf(grpc::Status, magma::UPFSessionContextState response) {
+  std::string imsi = response.session_snaphot().seid();
+  imsi.replace(15, 7, "");
+  // imsi.replace(imsi.end()-6,imsi.end(),"");
+  const std::string session_id = response.session_snaphot().seid();
+  MLOG(MINFO) << " Async Response received from UPF " << imsi

```

Comment:
Yes,  in the course of change as well.  I ll revisit in my next commit

Reply:
Why does imsi needs to be stripped?

`imsi.replace(15, 7, "");`

Reply:
The reason was -- initially we thought sessionid we use for "Seid", now we changed that to "imsi" + "local te_id".  currently -- we are already sending "session-id"  so we are using. session id is combination of "imsi " + 6 bytes. We are stripping last 6 bytes.  Anyway  this will be changed --- as we moved to imsi+ local te_id.  Accordinly session search routines also need to be extended.

Reply:
If you need to recover the IMSI I would strongly recommend to send the IMSI itself and receive it back as a field itself. If in the future we change session_id format, there is no way to know we need to change it here too.